src/api/user.service.ts

```diff
@@ -61,6 +66,16 @@ class UserService {
     // Uncomment this to make the actual API call
     // return this.api.delete<UserDeletionResponse>(`${this.controller}/${id}`)
   }
+
+  async updateProfile(body: ProfileUpdateRequest) {
+    return new Promise<UpdateProfileResponse>((resolve) => {
+      setTimeout(() => {
+        resolve(mockUpdateProfileResponse(body) as UpdateProfileResponse)
+      }, 500)
+    })
+    // Uncomment this to make the actual API call
+    // return this.api.put<ProfileUpdateResponse>(`${this.controller}/profile`, body);
```

Comment:
this should be userService.updatePorfile() we should avoid using direct api class